File: deck/base/code/native/node/runtime/bit.ts
// Bitwise integer operations over the node host. The JavaScript bitwise operators are 32-bit; keep operands within 32
// bits on this target. Reached only through the public bit API.
const bit = {
  and: (left: number, right: number): number => left & right,
  or: (left: number, right: number): number => left | right,
  exclusiveOr: (left: number, right: number): number => left ^ right,
  not: (value: number): number => ~value,
  shiftLeft: (value: number, count: number): number => value << count,
  shiftRight: (value: number, count: number): number => value >> count,
}

File: deck/base/code/native/swift/runtime/bit.swift
// Bitwise integer operations over swift (full 64-bit Int). Reached only through the public bit API.
enum bit {
    static func and(_ left: Int, _ right: Int) -> Int { return left & right }
    static func or(_ left: Int, _ right: Int) -> Int { return left | right }
    static func exclusiveOr(_ left: Int, _ right: Int) -> Int { return left ^ right }
    static func not(_ value: Int) -> Int { return ~value }
    static func shiftLeft(_ value: Int, _ count: Int) -> Int { return value << count }
    static func shiftRight(_ value: Int, _ count: Int) -> Int { return value >> count }
}
